### PR TITLE
Host Level Queue Task Processor

### PR DESCRIPTION
**Pull request by yycptt**

### CODE REVIEW

File: service/history/queueTaskProcessor.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/task"
+)
+
+type (
+	queueTaskProcessorOptions struct {
+		fifoSchedulerOptions *task.FIFOTaskSchedulerOptions
```
Comment (yux0) [inline service/history/queueTaskProcessor.go]: What about having a type here and use switch to replace those if statement?

File: service/history/queueTaskProcessor.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	if err != nil {
+		return false, err
+	}
+	return scheduler.TrySubmit(task)
+}
+
+func (p *queueTaskProcessorImpl) prepareSubmit(
+	task queueTask,
+) (task.Scheduler, error) {
+	if !p.isRunning() {
```
Comment (yux0) [inline service/history/queueTaskProcessor.go]: As discussed, this check could move to getOrCreateTaskScheduler

File: service/history/queueTaskProcessor.go (hunk 1 above)
Comment (yycptt) [reply service/history/queueTaskProcessor.go]: Done

File: service/history/queueTaskProcessor.go (hunk 2 above)
Comment (yycptt) [reply service/history/queueTaskProcessor.go]: Done.

**Pull request merged**